toolchain/check/testdata/impl/no_prelude/import_interface_assoc_const.carbon

```diff
@@ -96,6 +96,52 @@ impl C5 as I3 where .T1 = BAD1 and .T2 = {.a: {}} and .T3 = BAD2;
 // CHECK:STDERR:
 impl C5 as I3 where .T1 = {.b: {}} and .T2 = BAD3 and .T3 = BAD4 { }
 
+// --- fail_mismatch_mismatch_value.carbon
+library "[[@TEST_NAME]]";
+import library "interface";
+
+class CD { }
+class B1 { }
+class B2 { }
+class B3 { }
+class B4 { }
+class B5 { }
+class B6 { }
+class B7 { }
+class B8 { }
+class G1 { }
+class G2 { }
+
+// This is testing that it won't complain about mismatching values between
+// impl redeclarations if one of them is an error. We use mismatching values
+// for an associated constant within a single impl declaration in order to
+// create an error that only affects that member instead of the whole facet
+// type.
+
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE+8]]:12: error: associated constant T3 given two different values [AssociatedConstantWithDifferentValues]
+// CHECK:STDERR: impl CD as I3 where .T1 = B1 and .T1 = B2 and .T2 = G1 and .T3 = B3 and .T3 = B4;
+// CHECK:STDERR:            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE+4]]:12: error: associated constant T1 given two different values [AssociatedConstantWithDifferentValues]
+// CHECK:STDERR: impl CD as I3 where .T1 = B1 and .T1 = B2 and .T2 = G1 and .T3 = B3 and .T3 = B4;
+// CHECK:STDERR:            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+impl CD as I3 where .T1 = B1 and .T1 = B2 and .T2 = G1 and .T3 = B3 and .T3 = B4;
+
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE+12]]:12: error: associated constant T3 given two different values [AssociatedConstantWithDifferentValues]
+// CHECK:STDERR: impl CD as I3 where .T1 = G2 and .T2 = B5 and .T2 = B6 and .T3 = B7 and .T3 = B8 { }
+// CHECK:STDERR:            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE+8]]:12: error: associated constant T2 given two different values [AssociatedConstantWithDifferentValues]
+// CHECK:STDERR: impl CD as I3 where .T1 = G2 and .T2 = B5 and .T2 = B6 and .T3 = B7 and .T3 = B8 { }
+// CHECK:STDERR:            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_mismatch_mismatch_value.carbon:[[@LINE-10]]:1: error: impl declared but not defined [ImplMissingDefinition]
+// CHECK:STDERR: impl CD as I3 where .T1 = B1 and .T1 = B2 and .T2 = G1 and .T3 = B3 and .T3 = B4;
+// CHECK:STDERR: ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// CHECK:STDERR:
```

Comment:
Same comment as for the corresponding case in `toolchain/check/testdata/impl/no_prelude/impl_assoc_const.carbon`.

Reply:
Delete this one too.